### PR TITLE
Multiple callbacks to the same sequence

### PR DESCRIPTION
**Pull request by pennh**
This fix is based on issue #304: [https://github.com/ccampbell/mousetrap/issues/304]

Added append parameter to bind functions.
Added removeCallback parameter to unbind and _getMatched functions.
Separated the _unbind from the _bindeSingle function.

### CODE REVIEW

**Comment by pennh** [comment]
Hi,
@vicary is this fulfill your design?
@robincafolla and @ccampbell can you please also review?

Regards,
pennh

File: mousetrap.js [hunk 1]
```diff
@@ -140,6 +140,12 @@
     var _REVERSE_MAP;
 
     /**
+     * An empty function
+     *
+     * @type {Function}
+     */
+    var _EMPTY_FUNCTION = function() {};
+    /**
```
Comment (robincafolla) [inline mousetrap.js:149]: newline. #nit

File: mousetrap.js [hunk 2]
```diff
@@ -587,7 +594,8 @@
                     // @todo make deleting its own method?
                     var deleteCombo = !sequenceName && callback.combo == combination;
                     var deleteSequence = sequenceName && callback.seq == sequenceName && callback.level == level;
-                    if (deleteCombo || deleteSequence) {
+                    var deleteSpecificCallback = !removeCallback || removeCallback && callback.callback === removeCallback;
```
Comment (robincafolla) [inline mousetrap.js]: Explicit brackets please

```
deleteSpecificCallback = !removeCallback || (removeCallback && callback.callback === removeCallback);
```

File: mousetrap.js [hunk 3]
```diff
@@ -1031,4 +1111,4 @@
             return Mousetrap;
         });
     }
-}) (window, document);
+}) (window, document);
```
Comment (robincafolla) [inline mousetrap.js]: Keep the newline at the end please.

File: tests/test.mousetrap.js [hunk 4]
```diff
@@ -168,6 +168,36 @@ describe('Mousetrap.bind', function() {
...
+            KeyEvent.simulate('a'.charCodeAt(0), 65);
+            expect(spy1.callCount).to.equal(1, 'callback should fire for lowercase a');
+
+            var spy2 = sinon.spy();
+            Mousetrap.bind('a', spy2, undefined, false);
+
+            KeyEvent.simulate('a'.charCodeAt(0), 65);
+            expect(spy1.callCount).to.equal(1, 'callback should not fire for a first callback');
+            expect(spy2.callCount).to.equal(1, 'callback should fire for a second callback');
+        });
```
Comment (robincafolla) [inline tests/test.mousetrap.js:200]: Can you add a test for bind with `append` absent too?

**Comment by robincafolla** [comment]
@pennh looks good, fix the minor nits and add that extra test and I'll merge it.

**Comment by pennh** [comment]
Hi,
@robincafolla I added the fixes and additional test.
@vicary it will be great if you can also review it.

Regards,
@pennh

**Comment by robincafolla** [comment]
I've merged that and issued a new release v1.6.0 with it included.

Just realized that this isn't documents. @pennh would you mind PRing the README?

**Comment by vicary** [comment]
lgtm, thanks for making this happen.

**Pull request merged**